cmd/commands/root.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"github.com/spf13/cobra"
+)
+
+var (
+	KubeConfig           string
+	OperatorNamespace    string
+	CephClusterNamespace string
+)
+
+// rookCmd represents the rook command
+var RootCmd = &cobra.Command{
+	Use:   "rook-ceph",
+	Short: "kubectl rook-ceph provides common management and troubleshooting tools for Ceph.",
+	Args:  cobra.MinimumNArgs(1),
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	cobra.CheckErr(RootCmd.Execute())
+}
+
+func init() {
+
+	// Define your flags and configuration settings.
+
+	RootCmd.PersistentFlags().StringVar(&KubeConfig, "kubeconfig", "", "kubernetes config path")
+	RootCmd.PersistentFlags().StringVar(&OperatorNamespace, "opeator-namespace", "rook-ceph", "Kubernetes namespace where rook operator is running")
+	RootCmd.PersistentFlags().StringVar(&CephClusterNamespace, "namespace", "rook-ceph", "Kubernetes namespace where ceph cluster is created")
```

Comment:
How the krew cmd will run can you give a doc example for ceph health,

I can't how you might be calling the status, does it like calling the k8s client and then calling ceph cmd through it?

And one more question/doubt What if we just create and call the cmds from toolbox and wrap them into a go package?

Reply:
Currently, we are using operator pod for running ceph commands and toolbox and operator pod are almost same. So we prefer rook operator pod as rook/rook also uses operator pod for running ceph command internally